lib/routes/application.js

```diff
@@ -0,0 +1,121 @@
+const express = require('express');
+const {applicationPost, applicationGet, getLatestVersion, applicationGetFile} = require('../services/application')
+const fs = require('fs');
+path = require('path');
+const {promisify}=require('util')
+const stat = promisify(fs.stat)
+
+const app = () => {
+    const router = express.Router();
+    
+    router.get('/:name/list', async (req, res, next) => {
+        try {
+            let name = req.params.name;   
+            if (!name) {
+                res.status(404)
+                res.end();
+            } else {
+                name = name.toLowerCase()
+                ret = await applicationGet(name)
+                if (!ret){
+                    res.status(404)
+                    res.end();
```

Comment:
res.end() doesn't return. the execution continues to the next line.
either do return res.end() or return on the next line